tests/robot-tests/tests/libs/common.robot

```diff
@@ -369,9 +375,13 @@ user waits until h3 is visible
     [Arguments]   ${text}  ${wait}=${timeout}
     user waits until element is visible  xpath://h3[text()="${text}"]  ${wait}
 
+user waits until legend is visible
+    [Arguments]   ${text}  ${wait}=${timeout}
+    user waits until element is visible  xpath://legend[text()="${text}"]  ${wait}
+
 user waits until page contains title
     [Arguments]   ${text}  ${wait}=${timeout}
-    user waits until page contains element   xpath://h1[@data-testid="page-title" and text()="${text}"]   ${wait}
+    user waits until page contains element   xpath://h1[@data-testid="page-title ${text}" and text()="${text}"]   ${wait}
```

Comment:
If the data-testid now contains the page title text as well, it seems redundant to also check the text itself. My inclination would be change the testid in the frontend so it is just `page-title` again. Not sure why it has been added.

In most of the tests we're using `user waits until h1 is visible` to check the title. Not part of this PR, but it seems like there is potential cleanup where we could use `user waits until page contains title` instead

Reply:
Agreed - removed the text from the testid as I didn't see it being used for any other purpose.